src/javascript/app_2/Stores/ui-store.js

```diff
@@ -243,4 +237,16 @@ export default class UIStore extends BaseStore {
     setHasOnlyForwardingContracts(has_only_forward_starting_contracts) {
         this.has_only_forward_starting_contracts = has_only_forward_starting_contracts;
     }
+
+    @action.bound
+    addNotificationBar(message) {
+        if (!this.push_notifications.length) {
+            this.push_notifications.push(message);
+        }
+        this.push_notifications.map(notification => {
+            if (notification.msg_type !== message.msg_type) {
```

Comment:
I think to avoid duplicates you need to do a find instead, since in mapping if the message.msg_type of the first notification is different but the second one is the same, it never reaches the check for the second one and just pushes it into the array after the first check passes.